direct/src/distributed/ServerRepository.py

```diff
@@ -372,20 +374,33 @@ def handleDatagram(self, datagram):
         elif type == CLIENT_OBJECT_UPDATE_FIELD_TARGETED_CMU:
             self.handleClientObjectUpdateField(datagram, dgi, targeted = True)
         elif type == OBJECT_DELETE_CMU:
-            self.handleClientDeleteObject(datagram, dgi.getUint32())
+            self.handleClientDeleteObject(datagram, dgi)
         elif type == OBJECT_SET_ZONE_CMU:
             self.handleClientObjectSetZone(datagram, dgi)
         else:
             self.handleMessageType(type, dgi)
 
     def handleMessageType(self, msgType, di):
         self.notify.warning("unrecognized message type %s" % (msgType))
+    
+    def disconnectClient(self, connection):
+        """ disconnects the client """
+        client = self.clientsByConnection[connection]
+        
+        del self.clientsByConnection[connection]
+        del self.clientsByDoIdBase[client.doIdBase]
+        self.qcr.removeConnection(connection)
 
     def handleClientCreateObject(self, datagram, dgi):
         """ client wants to create an object, so we store appropriate
         data, and then pass message along to corresponding zones """
 
         connection = datagram.getConnection()
+        
+        if dgi.getRemainingSize() < 3:
```

Comment:
Still exploitable. Since `getUint32` reads 4 bytes and `getUint16` reads 2, you need 10 bytes.